src/infra/src/errors/mod.rs

```diff
@@ -110,12 +112,24 @@ pub enum PutDashboardError {
     MissingTitle,
     #[error("error putting dashboard with missing owner")]
     MissingOwner,
+    #[error("error putting dashboard with missing owner")]
+    AlertDestEmptyTemplateId,
```

Comment:
I originally wrote `PutDashboardError` type as being intended to expose just the possible errors that can occur when trying to perform a "Put Dashboard" operation. I assume this new `AlertDestEmptyTemplateId` variant is related to the "Put Destination" operation, so I'd recommend either renaming `PutDashboardError` to something more generic or moving the `AlertDestEmptyTemplateId` variant into its own `PutDestinationError` type. (Personally I'd recommend the second option, but both could work and that's a somewhat subjective/aesthetic opinion based on how I personally like to split up types.)

Reply:
ahh, yes. the second option was actually the intend. I tried to follow the same pattern but somehow put this variant into the DashboardError instead of DestinationError where it should've been. Thanks for catching it!